craft_parts/errors.py

```diff
@@ -81,3 +81,18 @@ def __init__(self, arch_name: str):
         resolution = "Make sure the architecture name is correct."
 
         super().__init__(brief=brief, resolution=resolution)
+
+
+class PartSpecificationError(PartsError):
+    """Parts were not correctly specified.
+
+    :param message: The error message.
+    """
+
+    def __init__(self, part_name: str, message: str):
```

Comment:
I have a strong personal preference for kwargs when the args are not intuitive :D 
```suggestion
    def __init__(self, *, part_name: str, message: str):
```